sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuDeviceManager.scala

```diff
@@ -203,8 +203,15 @@ object GpuDeviceManager extends Logging {
       var init = RmmAllocationMode.CUDA_DEFAULT
       val features = ArrayBuffer[String]()
       if (conf.isPooledMemEnabled) {
-        init = init | RmmAllocationMode.POOL
-        features += "POOLED"
+        if (conf.isPooledArenaEnabled) {
+          init = init | RmmAllocationMode.ARENA
+          features += "ARENA"
+        } else {
+          init = init | RmmAllocationMode.POOL
+          features += "POOLED"
+        }
+      } else if (conf.isPooledArenaEnabled) {
+        throw new IllegalStateException("Pooling arena is enabled but pooling is not")
```

Comment:
Do we want this behavior?  Sometimes users will want to disable all RMM pooling by simply toggling `spark.rapids.memory.gpu.pooling.enabled` and needing to _also_ toggle `spark.rapids.memory.gpu.pooling.arena.enabled` is kind of annoying.

My previous comment on throwing a clear error description was more targeted at the behavior where pooling.enabled meant use RMM default pooling and arena.enabled meant use ARENA.  Using both at the same time isn't allowed, and therefore we'd need a clear error message.  Now that the arena enable is a sub config of pooling, trying to use both at the same time cannot happen.

Reply:
Yeah agree it would be nice to able to control the pool irrespective of what implements the pool. 

Could `spark.rapids.memory.gpu.pooling.enabled` turn the pool on and off outright, whether the strategy is arena or default (?)

And so then the other setting is: `spark.rapids.memory.gpu.pooling.strategy=arena` or `spark.rapids.memory.gpu.pooling.strategy=default`.

Reply:
Strategy sounds good. Done.

Reply:
why not use `spark.rapids.memory.gpu.pooling=none|default|arena` to represent `disable pool`, `pool_memory_resouce`, `arena_memory_resouce`?

Reply:
I do like the idea of using an enum for pooling.  It helps to make it clear the different layers involved in allocation. 

1. Base resource allocator (cuda vs managed memory)
2. The pooling sub allocator, if any
3. Then the different tracking/debugging layers on top of it.  Logging, etc.

It would probably be nice to have 3 configs, one for each conceptual layer with the correct types to be clear if you should pick one of N options, or  N out of M options.

It probably would be good to go back and update the cudf RMM API too, but at this point I think we could do it as a follow on issue.  Mostly because I would want to maintain backwards compatibility with the existing configs for now.

Reply:
The current RMM api allows this layering but doesn't enforce it.

I like @JustPlay's suggestion, but it does break backwards compatibility. Is that ok?

Reply:
I would prefer it if we could deprecate the old configs and have a good way to transition from one to the other, but if that is more work then it is worth I am OK with breaking backwards compatibility.

@jlowe what do you think?

Reply:
I am somewhat concerned about backwards compatibility since we told users in the past to explicitly run with `spark.rapids.memory.gpu.pooling.enabled=false` (e.g.: original xgboost+plugin use case).  It will be very surprising to those users when that config stops working in the new release.

Unfortunately we don't have a framework in place to deal with deprecated/removed configs as Spark's `SQLConf` does.  It would be nice to have, but I agree it's not something that should be part of this PR.  I filed #878 to track adding such a framework.

I'd prefer in the short-term marking `spark.rapids.memory.gpu.pooling.enabled` as a deprecated config in its documentation.  We still honor it in one specific case.  If it's set to false then we treat `spark.rapids.memory.gpu.pooling` (or whatever we're calling the new config) as if it is set to `none`, and emit a deprecation warning stating what is happening and that the enable config may cease to function in a subsequent release.

Does that sound reasonable?  Shouldn't be a lot of extra code and preserves compatibility with the advertised use cases that required setting the pooling enabled property to false.

Reply:
Done.